lib/crypto.js

```diff
@@ -652,6 +672,14 @@ function filterDuplicates(names) {
   }).sort();
 }
 
+// timingSafeEqual reuses the timing-safe Hmac.equals function (see above) for
+// comparison of non-hash inputs, like capability URLs or authentication tokens.
+exports.timingSafeEqual = function(a, b) {
+  var key = randomBytes(32);
+  var ah = new Hmac('sha256', key).update(a);
+  return ah.validate(new Hmac('sha256', key).update(b).digest());
```

Comment:
This looks like a stack of cards. There should be no dependencies in a function like this.